pkg/sfu/streamtracker/streamtracker_dd.go

```diff
@@ -0,0 +1,272 @@
+package streamtracker
+
+import (
+	"sync"
+	"time"
+
+	"go.uber.org/atomic"
+
+	"github.com/livekit/livekit-server/pkg/sfu/buffer"
+	dd "github.com/livekit/livekit-server/pkg/sfu/dependencydescriptor"
+)
+
+type StreamTrackerDependencyDescriptor struct {
+	lock             sync.RWMutex
+	paused           bool
+	generation       atomic.Uint32
+	params           StreamTrackerParams
+	maxSpatialLayer  int32
+	maxTemporalLayer int32
+
+	onStatusChanged    [buffer.DefaultMaxLayerSpatial + 1]func(status StreamStatus)
+	onBitrateAvailable [buffer.DefaultMaxLayerSpatial + 1]func()
+
+	lastBitrateReport time.Time
+	bytesForBitrate   [buffer.DefaultMaxLayerSpatial + 1][buffer.DefaultMaxLayerTemporal + 1]int64
+	bitrate           [buffer.DefaultMaxLayerSpatial + 1][buffer.DefaultMaxLayerTemporal + 1]int64
+
+	isStopped bool
+}
+
+func NewStreamTrackerDependencyDescriptor(params StreamTrackerParams) *StreamTrackerDependencyDescriptor {
+	return &StreamTrackerDependencyDescriptor{
+		params:           params,
+		maxSpatialLayer:  -1,
```

Comment:
For consistency, initialise to `buffer.InvalidLayerSpatial` and `buffer.InvalidLayerTemporal`.